edweiss-app/components/lectures/remotecontrol/abstractRmtCtl.tsx

```diff
@@ -28,24 +37,47 @@ interface AbstractRmtCrlProps {
     handleLeft: () => void;
     handleMic: () => void;
     isRecording: boolean;
+    lang: AvailableLangs;
+    setLang: (lang: AvailableLangs) => void;
+
 }
 
-export const AbstractRmtCrl: React.FC<AbstractRmtCrlProps & LightDarkProps> = ({ handleRight, handleLeft, handleMic, isRecording }) => {
+export const AbstractRmtCrl: React.FC<AbstractRmtCrlProps & LightDarkProps> = ({ handleRight, handleLeft, handleMic, isRecording, lang, setLang }) => {
+
+
+    const modalRefLangSelect = useRef<BottomSheetModal>(null);
+
     return (
         <>
             <RouteHeader disabled title={"Lecture's Slides"} />
 
             <TView borderColor='text' m={'md'} mt={'xl'} mb={'xl'} b={0.5} backgroundColor='base' radius={'lg'}>
-                <TView mt={60} mb={70} justifyContent='center' alignItems='center'>
-                    <TText size={25} mb={'md'}> {t(`showtime:showtime_title`)}</TText>
-                    <TText size={35} >{t(`showtime:rmt_cntl_title`)}</TText>
-                    {isRecording ? (
-                        <TText mt={15} size={15} color='red'> {t(`showtime:recording_start`)} </TText>
-                    ) : (
-                        <TText mt={15} size={15} color='green'> {t(`showtime:tap_to_start_recording`)} </TText>
-                    )}
+
+                {/* STRC text, set language and timer */}
+                <TView mt={17} mb={50} justifyContent='center' alignItems='center'>
```

Comment:
I honestly consider this hardcoded values for mt mb a little too much, since Youssef has defined some default values of predefined sizes that are more helpful for having a clear understanding of a code. And since mt{'md'} would be the same as writting mt{16} I think it will most likely not have any impact on the display given that the difference is only by 1. I insist on this because most of us are using these default parameters to mantain a certain level of consistency and also helps keep symetry of the display since it becomes easier to follow always the same rounded values.

I understand if in some cases the values are too large or too little for what you need but it might not be always the case, and if you use a lot of some other values you could also consider increasing the possible default values to your liking so that you have more options then just 5

Reply:
Hey there, beforehand I would like to stress the fact that you are entirely right about this point.
In fact, in the first version I have did it this way using the predefined set of values. 
However, I would like to explain you the reason for it existence and why I had no other choice but to do it.

I will go back in the development process that took me to go this `unconventional path`:

Before diving in I would like to argue why we did the `md`, `sm` in the frst place ? The reason for this as you refresh it is when you have different type of components `TTextInput` `TText` ... is to keep UI/UX consistency. However, in this very exceptional case, I am only using the same components and have fine tuned the values so that it feels the more adequate to the situation. 

Firstly, I have created a new `ApplicationRoute` with the boxes to select the languages. However, the boxes where not alligned (due to the `space-between`), this made the interspace between the 2 boxes uneven. So to overcome this I had to fix the size for `width` and `height`. However, now we got another problem is that the boxes are not spaced correctly anymore. That's why in the `TwoLangSelect` component you could say tht I coukd have refarctored a bit to make it more modular, but the very important part is the `mr` and `ml` depending on the 

After when it was intedended to fit in a modal, I had to ....

For this reason adding a 1 pixel tolerance makes it not as elegant (I know I am a bit perfectionist 😋). Moreover as you say, I could have also created default values. However, this is a very niche use case as it's intended to be used only within this modal.

However, I would like to point out that you made me think of an improvement:
For the `TwoLangSelect` component, we could have added a check for  `lang2` in order to make it displayable only if provided by the user. This way if the user adds an uneven amount of langs, we could still be able to display it in the modal. (This is indeed a bug as if someone remmove one language, the code will fail (not applying the separation of privillege principle 🙃))

Hope it's more clear. But yeah you are totally right, it's not a good practice but the circumstance, have made that I went this path. 

Are you okay with this ?

Reply:
Okay thank you for the thorough explanation